### PR TITLE
feat: Add ability to retrieve payment terms using external id

### PR DESCRIPTION
**Pull request by warmwaffles**
* Get a payment term using internal id or external id.
* Update a payment term using internal id or external id.
* Update a payment term's external id.
* Delete a payment term using internal id or external id.
* Create a payment with an external id. Upsert is not implemented, nor will it be.

### CODE REVIEW

File: vic.api.v0.yaml [hunk 1]
```diff
@@ -1774,14 +1774,37 @@ paths:
...
+          $ref: '#/components/responses/ForbiddenResponse'
+        '404':
+          $ref: '#/components/responses/ResourceNotFoundResponse'
     put:
       description: |
         Update a payment term.
+        You may use the `externalId` in the, but must pass `useSystem=external`
+        in order to fetch the payment term. You are permitted to update the
+        `externalId` as well when updating the payment term. Leaving it
+        unspecified **WILL** clear the `externalId`.
```
Comment (jonathanpglick) [inline vic.api.v0.yaml:1801]:
```suggestion
        The `id` in the path can be either the Payment Term's internal ID or its external ID
        when `useSystem=external` is included in the request.

        The Payment Term's external ID can be updated by including `externalId={value}`
        in the request body. If the `externalId` key is present without a value, the
        Payment Term's `externalId` **WILL** be cleared.
```

File: vic.api.v0.yaml [hunk 2]
```diff
@@ -1794,11 +1817,14 @@ paths:
     delete:
       description: |
         Delete a payment term.
+        You may use the `externalId` in the, but must pass `useSystem=external`
+        in order to fetch the payment term.
```
Comment (jonathanpglick) [inline vic.api.v0.yaml:1821]:
```suggestion
        The `id` in the path can be either the Payment Term's internal ID or its external ID
        when `useSystem=external` is included in the request.
```

**Review by jonathanpglick — CHANGES_REQUESTED** [review]
I think `UpdatePaymentTerm` and `PaymentTerm` objects are both missing `externalId`.

File: vic.api.v0.yaml (hunk 2 above)
Comment (warmwaffles) [reply vic.api.v0.yaml]: Done. Reused the same description you suggested for `getPaymentTerm`

File: vic.api.v0.yaml (hunk 1 above)
Comment (warmwaffles) [reply vic.api.v0.yaml]: Much better.

**Review by jonathanpglick — APPROVED** [review]

**Review by mbaleczny — APPROVED** [review]

**Pull request merged**